rust/datafusion/src/datasource/datasource.rs

```diff
@@ -24,6 +24,15 @@ use crate::arrow::datatypes::SchemaRef;
 use crate::error::Result;
 use crate::physical_plan::ExecutionPlan;
 
+/// The table statistics
+#[derive(Clone)]
+pub struct Statistics {
+    /// The number of table rows
+    pub num_rows: i64,
```

Comment:
```suggestion
    /// The number of table rows
    pub num_rows: u64,
```

Reply:
Is it better to use `usize`

Reply:
My understanding is that because the parquet reader produces` i64` which (assuming above 0) is less than `u64` whereas a `usize` _may_ be smaller depending on compile target.

Reply:
I think here use `usize` is good than u64, because  while we implement the interface we should check the max value  less than  MAX `usize` and the value  strongly dependent on compile target.

Reply:
I think `usize` is good enough to begin with -- if we need to change it ot u64 for some reason in the future (e.g. to support > 4GB files on 32-bit architectures) that should be a straightforward change.